### PR TITLE
Register layout screen created

### PR DESCRIPTION
**Pull request by miguelhenaoa**
## Summary

- Created SignUp screen
- Created form component and use ng-content
- Console log to form value

## Trello Card
https://trello.com/c/NmxsYJSj/4-register-layout

## Screenshot
Captura de Pantalla 2021-05-24 a la(s) 8 50 56 a  m

### CODE REVIEW

File: src/app/components/form/form.component.html [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<div class="div-flex content">
+  <div class="form-content">
```
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.html]: This div it's not necessary

File: src/app/components/form/form.component.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    <div class="div-flex image-brand">
```
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.html]: Same here

File: src/app/components/form/form.component.html [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+      <img class="logo-wolox" src="assets/images/logo_wolox.svg">
```
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.html]: Add alt property

File: src/app/components/form/form.component.scss [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  width: inherit;
+
+  .logo-wolox {
+    height: 70px;
+    width: 212.42px;
+  }
+}
+
+
+
```
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.scss]: Just leave one empty line.

File: src/app/components/form/form.component.ts [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+@Component({
+  selector: 'app-form',
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.scss']
+})
+export class FormComponent implements OnInit {
+
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.ts]: Delete methods you don't use

File: src/app/screens/sign-up/sign-up-routing.module.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+const routes: Routes = [
+  { path: '', component: SignUpComponent }
+];
+
+@NgModule({
+  imports: [RouterModule.forChild(routes)],
+  exports: [RouterModule]
+})
+export class SignUpRoutingModule { }
```
Comment (ebedoyaalzate) [inline src/app/screens/sign-up/sign-up-routing.module.ts:13]: Do you really need routing module for Sign-up module?

File: src/app/screens/sign-up/sign-up.component.html [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+      <label for="password">Password</label>
+      <input type="password" [(ngModel)]="signUpForm.password" [ngModelOptions]="{ standalone: true}">
+    </div>
+
+    <div class="content-input">
+      <label for="confirmPassword">Confirmación de Password</label>
+      <input type="password" [(ngModel)]="signUpForm.confirmPassword" [ngModelOptions]="{ standalone: true}">
+    </div>
+
+    <div class="section-buttons">
```
Comment (ebedoyaalzate) [inline src/app/screens/sign-up/sign-up.component.html]: This div it's not necessary.

File: src/app/screens/sign-up/sign-up.component.html [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+      <button class="btn-fill" type="submit">Sign Up</button>
+      <div class="line-button"></div>
```
Comment (ebedoyaalzate) [inline src/app/screens/sign-up/sign-up.component.html]: Remember to use as less html code as possible for clarity and reusability. Try to use the decorative part via css only and just the structure and functionality parts in the html, for this case you can use pseudo elements.

File: src/app/screens/sign-up/sign-up.component.html [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+    <div class="content-input">
+      <label for="password">Password</label>
+      <input type="password" [(ngModel)]="signUpForm.password" [ngModelOptions]="{ standalone: true}">
+    </div>
+
+    <div class="content-input">
+      <label for="confirmPassword">Confirmación de Password</label>
+      <input type="password" [(ngModel)]="signUpForm.confirmPassword" [ngModelOptions]="{ standalone: true}">
+    </div>
```
Comment (ebedoyaalzate) [inline src/app/screens/sign-up/sign-up.component.html]: This html it's too repetitive, you can use ng-for for a more clean code

File: src/app/screens/sign-up/sign-up.component.ts [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+  selector: 'app-sign-up',
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.scss'],
+  providers: [FormBuilder]
+})
+export class SignUpComponent implements OnInit {
+
+  signUpForm: SignUp = { name: '', lastname: '', email: '', password: '', confirmPassword: '' };
+
+  constructor() {
```
Comment (ebedoyaalzate) [inline src/app/screens/sign-up/sign-up.component.ts]: remove the methods you don't use.

File: src/styles/base/buttons.scss [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  font-family: $font-family;
+  font-size: $font-size;
+  height: 32px;
+  width: 252px;
+}
+
+
```
Comment (ebedoyaalzate) [inline src/styles/base/buttons.scss]: Just leave one line empty.

File: src/app/components/form/form.component.html [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+    </div>
+
+    <div class="form">
+      <ng-content></ng-content>
+    </div>
+  </div>
+</div>
```
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.html]: Add empty line.

File: src/app/components/form/form.component.scss (hunk 4 above)
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.scss]: Just leave one line empty.

Comment (ebedoyaalzate) [inline src/app/components/form/form.component.scss]: Just leave one empty line.

File: src/app/components/form/form.component.html (hunk 12 above)
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.html]: Add empty line.

File: src/styles/styles.scss [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+body {
+  margin: 0;
+  padding: 0;
+}
+
+.div-flex {
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: center;
+}
```
Comment (ebedoyaalzate) [inline src/styles/styles.scss:10]: In styles.scss just import files.

File: src/styles/base/form.scss [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+label {
+  height: 22px;
+  width: 100%;
+}
+
+input,
+label {
+  font-family: $font-family;
+  font-size: $font-size;
+}
```
Comment (ebedoyaalzate) [inline src/styles/base/form.scss]: Add empty line

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: src/app/app.module.ts [hunk 15]
```diff
@@ -3,14 +3,17 @@ import { BrowserModule } from '@angular/platform-browser';
...
   declarations: [
-    AppComponent
+    AppComponent,
   ],
   imports: [
     BrowserModule,
-    AppRoutingModule
+    AppRoutingModule,
+    FormsModule,
+    ReactiveFormsModule
```
Comment (CrisVega08) [inline src/app/app.module.ts]: Do you need these here?

File: src/app/components/form/form.component.scss [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+.form-content {
+  background-color: $color-grey;
+  border-top: 8px solid $color-brand-blue;
+  margin-top: 50px;
+  max-width: 100%;
+  width: $width-form;
+  
+  .form {
+    margin: 24px;
+    max-width: 252px;
```
Comment (CrisVega08) [inline src/app/components/form/form.component.scss]:
```suggestion
    max-width: 250px;
```

Remember to use a base value, can be 5px

File: src/app/components/form/form.component.scss [hunk 17]
```diff
@@ -0,0 +1,34 @@
...
+
+.form-content {
+  background-color: $color-grey;
+  border-top: 8px solid $color-brand-blue;
+  margin-top: 50px;
+  max-width: 100%;
+  width: $width-form;
+  
+  .form {
+    margin: 24px;
```
Comment (CrisVega08) [inline src/app/components/form/form.component.scss]:
```suggestion
    margin: 25px;
```

File: src/app/components/form/form.component.scss [hunk 18]
```diff
@@ -0,0 +1,34 @@
...
+    max-width: 252px;
+    width: 100%;
+  }
+}
+
+.image-brand {
+  margin-top: 28px;
```
Comment (CrisVega08) [inline src/app/components/form/form.component.scss]:
```suggestion
  margin-top: 30px;
```

File: src/app/components/form/form.component.scss [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+  max-width: inherit;
+  width: inherit;
+
+  .logo-wolox {
+    height: 70px;
+    width: 212.42px;
```
Comment (CrisVega08) [inline src/app/components/form/form.component.scss]:
```suggestion
    width: 210px;
```

File: src/app/components/form/form.component.ts [hunk 20]
```diff
@@ -0,0 +1,15 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-form',
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.scss']
+})
+export class FormComponent implements OnInit {
+
+  constructor() { }
```
Comment (CrisVega08) [inline src/app/components/form/form.component.ts]: Also delete constructor if you aren't using

File: src/app/components/form/form.module.ts [hunk 21]
```diff
@@ -0,0 +1,12 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { FormComponent } from './form.component';
+
+@NgModule({
+  declarations: [FormComponent],
+  exports: [FormComponent],
+  imports: [
+    CommonModule
```
Comment (CrisVega08) [inline src/app/components/form/form.module.ts]: here you can import reactive forms and remove from app.module

File: src/app/screens/sign-up/interfaces/sign-up.interface.ts [hunk 22]
```diff
@@ -0,0 +1,7 @@
+export interface SignUp {
+  name: string;
+  lastname: string;
+  email: string;
+  password: string;
+  confirmPassword: string;
+}
```
Comment (CrisVega08) [inline src/app/screens/sign-up/interfaces/sign-up.interface.ts]: Add empty line

File: src/app/screens/sign-up/sign-up.component.scss [hunk 23]
```diff
@@ -0,0 +1,17 @@
+@import "src/styles/styles.scss";
+
+.line-button {
+  background-color: $color-grey-line;
+  height: 2px;
+  margin: 16px 0;
```
Comment (CrisVega08) [inline src/app/screens/sign-up/sign-up.component.scss]:
```suggestion
  margin: 15px 0;
```

File: src/app/screens/sign-up/sign-up.component.scss [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+  max-width: inherit;
+  width: inherit;
+}
+
+.section-buttons {
+  display: flex;
+  flex-flow: column nowrap;
+  justify-items: center;
+  margin: 24px 0;
```
Comment (CrisVega08) [inline src/app/screens/sign-up/sign-up.component.scss]:
```suggestion
  margin: 25px 0;
```

File: src/styles/base/buttons.scss [hunk 25]
```diff
@@ -0,0 +1,29 @@
+@import "/src/styles/lib/colors.scss";
```
Comment (CrisVega08) [inline src/styles/base/buttons.scss]: use partials for base `_colors.scss`

File: src/app/components/form/form.component.scss [hunk 26]
```diff
@@ -0,0 +1,34 @@
+@import "src/styles/styles.scss";
+
+.content {
+  max-width: 2000px;
+  width: 100%;
+}
+
+.form-content {
+  background-color: $color-grey;
```
Comment (CrisVega08) [inline src/app/components/form/form.component.scss]: Normally for colors name we use https://chir.ag/projects/name-that-color

File: src/app/components/form/form.component.scss [hunk 27]
```diff
@@ -0,0 +1,34 @@
...
+  border-top: 8px solid $color-brand-blue;
```
Comment (CrisVega08) [inline src/app/components/form/form.component.scss]: Review color name

File: src/app/screens/sign-up/sign-up.component.ts [hunk 28]
```diff
@@ -0,0 +1,25 @@
+import { Component, OnInit } from '@angular/core';
+import { FormBuilder } from '@angular/forms';
+import { SignUp } from './interfaces/sign-up.interface';
+
+@Component({
+  selector: 'app-sign-up',
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.scss'],
+  providers: [FormBuilder]
```
Comment (CrisVega08) [inline src/app/screens/sign-up/sign-up.component.ts]: why do you provide formBuilder?

File: src/styles/base/buttons.scss [hunk 29]
```diff
@@ -0,0 +1,29 @@
...
+@import "/src/styles/lib/variables.scss";
+
+/*
+* Buttons
+*/ 
+
+.btn-border {
+  background-color: $color-grey;
```
Comment (CrisVega08) [inline src/styles/base/buttons.scss]: Remember colors name

File: src/styles/base/buttons.scss [hunk 30]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+button {
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  font-family: $font-family;
+  font-size: $font-size;
+  height: 32px;
+  width: 252px;
```
Comment (CrisVega08) [inline src/styles/base/buttons.scss]:
```suggestion
  width: 250px;
```

File: src/styles/base/buttons.scss [hunk 31]
```diff
@@ -0,0 +1,29 @@
...
+  color: $color-white;
+}
+
+button {
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  font-family: $font-family;
+  font-size: $font-size;
+  height: 32px;
```
Comment (CrisVega08) [inline src/styles/base/buttons.scss]:
```suggestion
  height: 30px;
```

File: src/styles/base/form.scss [hunk 32]
```diff
@@ -0,0 +1,30 @@
+@import "/src/styles/lib/colors.scss";
+@import "/src/styles/lib/variables.scss";
+
+.content-input {
+  display: flex;
+  flex-flow: column nowrap;
+  margin: 16px 0;
```
Comment (CrisVega08) [inline src/styles/base/form.scss]:
```suggestion
  margin: 15px 0;
```

File: src/styles/base/form.scss [hunk 33]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+input {
+  background-color: $color-white;
+  border: none;
+  border-radius: 5px;
+  height: 32px;
+  margin-top: 8px;
+  width: 100%;
+  max-width: 248px;
```
Comment (CrisVega08) [inline src/styles/base/form.scss]:
```suggestion
  max-width: 250px;
```

File: src/styles/lib/colors.scss [hunk 34]
```diff
@@ -0,0 +1,10 @@
+/*
```
Comment (CrisVega08) [inline src/styles/lib/colors.scss]: _colors.scss

File: src/styles/lib/variables.scss [hunk 35]
```diff
@@ -0,0 +1,8 @@
+/*
```
Comment (CrisVega08) [inline src/styles/lib/variables.scss]: _variables

File: src/styles/lib/variables.scss [hunk 36]
```diff
@@ -0,0 +1,8 @@
...
+* Variables
+*/
+
+$width-form: 300px;
+
+$font-size: 16px;
+$font-family: 'Roboto', sans-serif;
```
Comment (CrisVega08) [inline src/styles/lib/variables.scss]: In this project you don't change fonts so use only once globally.

File: src/styles/styles.scss [hunk 37]
```diff
@@ -0,0 +1,17 @@
...
+@import "./lib/variables";
+@import "./base/buttons";
+@import "./base/form";
+
+body {
+  margin: 0;
+  padding: 0;
+}
+
+.div-flex {
```
Comment (CrisVega08) [inline src/styles/styles.scss]: Remove this clase and move to correspondently file

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/screens/sign-up/sign-up.component.ts (hunk 28 above)
Comment (miguelhenaoa) [reply src/app/screens/sign-up/sign-up.component.ts]: Error resolved when adding formBuilder as providers, but for now the class is not used, I will remove it.

File: src/app/screens/sign-up/sign-up.component.html [hunk 38]
```diff
@@ -0,0 +1,9 @@
+<app-form [buttonLabel]="'Login'" [buttonRoute]="'/login'">
+  <form class="form" (ngSubmit)="signUp()">
+    <div class="content-input" *ngFor="let field of fields">
+      <label [for]="field.name">{{field.label}}</label>
+      <input [type]="field.type" [(ngModel)]="signUpForm[field.name]" [ngModelOptions]="{ standalone: true}">
+    </div>
+    <button class="btn-fill" type="submit">Sign Up</button>
+  </form>
+</app-form>
```
Comment (ebedoyaalzate) [inline src/app/screens/sign-up/sign-up.component.html]: Add empty line.

File: src/app/screens/sign-up/sign-up.component.ts [hunk 39]
```diff
@@ -0,0 +1,23 @@
+import { Component, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-sign-up',
+  templateUrl: './sign-up.component.html',
+  styleUrls: ['./sign-up.component.scss']
+})
+export class SignUpComponent {
+
+  fields = [
```
Comment (ebedoyaalzate) [inline src/app/screens/sign-up/sign-up.component.ts]: Pass this object to a constants file.

File: src/styles/base/buttons.scss [hunk 40]
```diff
@@ -0,0 +1,27 @@
+@import "/src/styles/lib/colors.scss";
+@import "/src/styles/lib/variables.scss";
+
+/*
+* Buttons
+*/
```
Comment (ebedoyaalzate) [inline src/styles/base/buttons.scss]: you can omit this comments, with the file name it's fine.

File: src/styles/lib/_colors.scss [hunk 41]
```diff
@@ -0,0 +1,10 @@
+/*
+*  Branding Colors
+*/
```
Comment (ebedoyaalzate) [inline src/styles/lib/_colors.scss]: Same here.

File: src/styles/lib/_variables.scss [hunk 42]
```diff
@@ -0,0 +1,5 @@
+/*
+* Variables
+*/
```
Comment (ebedoyaalzate) [inline src/styles/lib/_variables.scss]: and here.

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: src/app/components/form/form.component.scss [hunk 43]
```diff
@@ -0,0 +1,27 @@
+@import "src/styles/styles.scss";
+
+button {
+  margin: 15px 25px 25px 25px;
```
Comment (CrisVega08) [inline src/app/components/form/form.component.scss]:
```suggestion
  margin: 15px 25px 25px;
```

File: src/app/screens/sign-up/sign-up.component.scss [hunk 44]
```diff
@@ -0,0 +1,5 @@
+@import "src/styles/styles.scss";
+
+button {
+  margin: 10px auto 15px auto;
```
Comment (CrisVega08) [inline src/app/screens/sign-up/sign-up.component.scss]:
```suggestion
  margin: 10px auto 15px;
```

File: src/styles/base/buttons.scss [hunk 45]
```diff
@@ -0,0 +1,23 @@
...
+.btn-fill {
+  background-color: $earls-green;
+  color: $white;
+}
+
+button {
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  font-family: 'Roboto', sans-serif;
```
Comment (CrisVega08) [inline src/styles/base/buttons.scss]: use universal selector in global style sheet

File: src/app/components/form/form.component.scss [hunk 46]
```diff
@@ -0,0 +1,27 @@
...
+  border-top: 8px solid $cerulean;
+  display: flex;
+  flex-flow: column nowrap;
+  margin: auto;
+  margin-top: 50px;
+  max-width: $width-form;
+  width: 100%;
+
+  .form {
+    border-bottom: 2px solid $silver-chalice;
```
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.scss]: Use pseudo elements for this

File: src/styles/base/form.scss [hunk 47]
```diff
@@ -0,0 +1,24 @@
+@import "/src/styles/lib/colors";
+@import "/src/styles/lib/variables";
+
+.content-input {
```
Comment (CrisVega08) [inline src/styles/base/form.scss]: I suggest move this to sing-up component, where you add de class

File: src/app/components/form/form.component.html [hunk 48]
```diff
@@ -0,0 +1,7 @@
+<div class="container">
+  <img class="logo-wolox" src="assets/images/logo_wolox.svg" alt="Logo Wolox">
+  <div class="form">
```
Comment (ebedoyaalzate) [inline src/app/components/form/form.component.html]: This div isn't necessary.

File: src/styles/styles.scss [hunk 49]
```diff
@@ -0,0 +1,13 @@
+/* You can add global styles to this file, and also import other style files */
+
+@import "./lib/colors";
```
Comment (CrisVega08) [inline src/styles/styles.scss]: Don import partials here

File: src/styles/styles.scss [hunk 50]
```diff
@@ -0,0 +1,13 @@
...
+@import "./lib/variables";
+@import "./base/buttons";
+@import "./base/form";
+
```
Comment (CrisVega08) [inline src/styles/styles.scss:5]: You can use universal selector for padding, margin and font

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/components/form/form.component.html [hunk 51]
```diff
@@ -0,0 +1,5 @@
+<div class="container">
+  <img class="logo-wolox" src="assets/images/logo_wolox.svg" alt="Logo Wolox">
+  <ng-content></ng-content>
+  <button class="btn btn-route" type="button" [routerLink]="buttonRoute">{{buttonLabel}}</button>
```
Comment (CrisVega08) [inline src/app/components/form/form.component.html]: How this is a routing you should use a link not button

File: src/index.html [hunk 52]
```diff
@@ -2,10 +2,12 @@
 <html lang="en">
 <head>
   <meta charset="utf-8">
-  <title>WoloxBooks</title>
+  <title>Wolox Books</title>
   <base href="/">
   <meta name="viewport" content="width=device-width, initial-scale=1">
   <link rel="icon" type="image/x-icon" href="favicon.ico">
+  <link rel="preconnect" href="https://fonts.gstatic.com">
+  <link href="https://fonts.googleapis.com/css2?family=Roboto&display=swap" rel="stylesheet">
```
Comment (CrisVega08) [inline src/index.html]: Do you need all font or only some parts ?

File: src/styles/base/buttons.scss [hunk 53]
```diff
@@ -0,0 +1,12 @@
+@import "/src/styles/lib/colors.scss";
```
Comment (CrisVega08) [inline src/styles/base/buttons.scss]: You don 't use colors here, delete this line.

File: src/styles/base/buttons.scss [hunk 54]
```diff
@@ -0,0 +1,12 @@
...
+@import "/src/styles/lib/variables.scss";
+
+.btn {
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  font-family: 'Roboto', sans-serif;
```
Comment (CrisVega08) [inline src/styles/base/buttons.scss]: if you use universal selector to add font, you don't need this line

File: src/styles/styles.scss [hunk 55]
```diff
@@ -0,0 +1,11 @@
+/* You can add global styles to this file, and also import other style files */
+
+@import "./base/buttons";
+@import "./base/form";
+
+* {
+  margin: 0;
+  padding: 0;
+  font-family: 'Roboto', sans-serif;
+  font-size: 16px;
```
Comment (CrisVega08) [inline src/styles/styles.scss]: remove.

File: src/styles/base/buttons.scss [hunk 56]
```diff
@@ -0,0 +1,12 @@
+@import "/src/styles/lib/colors.scss";
+@import "/src/styles/lib/variables.scss";
```
Comment (CrisVega08) [inline src/styles/base/buttons.scss]: Same here

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

**Review by ebedoyaalzate — APPROVED** [review]

**Review by CrisVega08 — APPROVED** [review]

**Pull request merged**